src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs

```diff
@@ -190,5 +187,146 @@ public static void ExtractToDirectory(string sourceArchiveFileName, string desti
+        /// - For entries where the language encoding flag is set, the entry names are decoded by using UTF-8.
+        /// If <paramref name="entryNameEncoding"/> is set to <see langword="null"/>, entry names are decoded according to the following rules:
+        /// - For entries where the language encoding flag (in the general-purpose bit flag of the local file header) is not set, entry names are decoded by using the current system default code page.
+        /// - For entries where the language encoding flag is set, the entry names are decoded by using UTF-8.
+        /// <exception cref="ArgumentException"><paramref name="destinationDirectoryName" />> is <see cref="string.Empty" />, contains only white space, or contains at least one invalid character.
+        /// -or-
+        /// <paramref name="entryNameEncoding"/> is set to a Unicode encoding other than UTF-8.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="destinationDirectoryName" /> or <paramref name="source" /> is <see langword="null" />.</exception>
+        /// <exception cref="PathTooLongException">The specified path in <paramref name="destinationDirectoryName" /> exceeds the system-defined maximum length.</exception>
+        /// <exception cref="DirectoryNotFoundException">The specified path is invalid (for example, it is on an unmapped drive).</exception>
+        /// <exception cref="IOException">The name of an entry in the archive is <see cref="string.Empty" />, contains only white space, or contains at least one invalid character.
+        /// -or-
+        /// Extracting an archive entry would create a file that is outside the directory specified by <paramref name="destinationDirectoryName" />. (For example, this might happen if the entry name contains parent directory accessors.)
+        /// -or-
+        /// <paramref name="overwriteFiles" /> is <see langword="false" /> and an archive entry to extract has the same name as an entry that has already been extracted or that exists in <paramref name="destinationDirectoryName" />.</exception>
+        /// <exception cref="UnauthorizedAccessException">The caller does not have the required permission to access the archive or the destination directory.</exception>
+        /// <exception cref="NotSupportedException"><paramref name="destinationDirectoryName" /> contains an invalid format.</exception>
+        /// <exception cref="InvalidDataException">The archive contained in the <paramref name="source" /> stream is not a valid zip archive.
+        /// -or-
+        /// An archive entry was not found or was corrupt.
+        /// -or-
+        /// An archive entry was compressed by using a compression method that is not supported.</exception>
+        public static void ExtractToDirectory(Stream source, string destinationDirectoryName, Encoding? entryNameEncoding, bool overwriteFiles)
+        {
+            ArgumentNullException.ThrowIfNull(source);
+            if (!source.CanRead)
```

Comment:
it will throw anyway

https://github.com/dotnet/runtime/blob/3747edb1812106f0b88a42b724c3e44b69dc79c4/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs#L122
https://github.com/dotnet/runtime/blob/3747edb1812106f0b88a42b724c3e44b69dc79c4/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs#L140